meilisearch/models/key.py

```diff
@@ -0,0 +1,49 @@
+from datetime import datetime
+from typing import List, Optional
+
+from camel_converter.pydantic_base import CamelBase
+
+
+class _KeyBase(CamelBase):
+    uid: str
+    name: Optional[str] = None
+    description: str
+    actions: List[str]
+    indexes: List[str]
+    expires_at: Optional[datetime] = None
+
+    class Config:
+        json_encoders = {
+            datetime: lambda v: None
+            if not v
+            else f"{str(v).split('.', maxsplit=1)[0].replace(' ', 'T')}Z"
+        }
+
+
+class Key(_KeyBase):
+    key: str
+    created_at: datetime
+    updated_at: Optional[datetime] = None
+
+
+class KeyUpdate(CamelBase):
+    key: str
+    name: Optional[str] = None
+    description: Optional[str] = None
+    actions: Optional[List[str]] = None
+    indexes: Optional[List[str]] = None
+    expires_at: Optional[datetime] = None
+
+    class Config:
+        json_encoders = {
+            datetime: lambda v: None
+            if not v
+            else f"{str(v).split('.', maxsplit=1)[0].replace(' ', 'T')}Z"
+        }
```

Comment:
If I understand correctly this class is only used for the return of the update_key method so why not use it directly Key? We can keep this class to provide a Class to update Key like:
```
class KeyUpdate(CamelBase):
    name: str = None
    description: str = None

def update_key(self, key_or_uid: str, options: KeyUpdate) -> Key:
```
But I'm not sure it's worth it. WDYT?

Reply:
This was a mistake, I should have used `Key` for the return. What I use `KeyUpdate` for is exactly what you described. The disadvantage is the user has to import `KeyUpdate`, but the advantage is the user knows exactly what to send.

I think an argument could be made for both options. I'll let you make the call if you want me to change `options` to `KeyUpdate`, or if you want me to remove the class.

Reply:
I don't have a strong opinion on this. I tend to agree with you, I find it easier for the user to keep the option. The only advantage of using a `KeyUpdate` is that there is no risk of making a mistake on how to update the key due to the Strong guidance.
Let's leave it as it is I think it's fine